in-progress/7588-spartan-clusters.md

```diff
@@ -0,0 +1,301 @@
+      action: 'kill'
+      actionOptions:
+        force: true
+```
+
+The `yarn-projects/+end-to-end-base` earthly image will be updated to include `kubectl`, `helm`, and `kind`.
+
+We will add new tests to `yarn-projects/end-to-end`, and expose them to CI as earthly targets in the form `yarn-projects/end-to-end/+network-{test_name}`.
+
+The `ci` github action workflow will have a new job analogous to `e2e` and `bench-e2e` called `network-e2e`. This job will be a matrix on the `+network` targets, and for each it will ensure the test machine has a k8s cluster running with the relevant docker images loaded (via kind), and run a particular earthly target.
+
+Each target will:
+1. deploy a network to the k8s cluster (via helm install)
+2. (optional) apply some chaos configurations
+3. run helm tests
+4. tear down the network
+
+
+## Implementation
+
+### Helm Chart
+
+**anvil**
+
+There will be a deployment for anvil. It will have a single replica, and be exposed via ClusterIP.
+
+**boot node**
+
+There will be a statefulset with a single replica for the boot node. As part of its init container it will deploy the enshrined L1 contracts. Other nodes in the network will be able to resolve the boot node's address via its stable DNS name, e.g. `boot-node-0.aztec-network.svc.cluster.local`.
+
+**full node**
+
+There will be a statefulset for the full nodes. The number of replicas will be configurable. Each full node will have a service exposing its p2p port and node port.
+
+As part of their init container, they will get config from the boot node, including its ENR (which will require exposing this on the `get-node-info` endpoint). 
+
+Tests will produce a validator set from the full nodes.
+
+It will be possible to address full nodes individually via their stable DNS name, e.g. `full-node-0.aztec-network.svc.cluster.local`, as well as collectively via a service, e.g. `full-node.aztec-network.svc.cluster.local`.
+
+**pxe**
+
+PXEs will be deployed as a statefulset. The number of replicas will be configurable. Each PXE will have a service exposing its port.
+
+PXEs will use the collective full node service as their node url by default.
+
+PXEs will only be able to be addressed individually.
+
+**opentel**
+
+There will be a deployment for opentel. It will have a single replica. Nodes in the network will push their logs to opentel.
+
+**prometheus**
+
+There will be a deployment for prometheus. It will have a single replica.
+
+**grafana**
+
+There will be a deployment for graphana. It will have a single replica, and be exposed via ClusterIP.
+
+
+### Staging Network
+
+We will create a long-lived k8s cluster deployed on AWS. This cluster will be used for running the `staging` network.
+
+We will use this network for long-running stress tests.
+
+There will be a github action workflow that deploys the network to this cluster on every push to the `staging` branch.
+
+The grafana dashboard will be exposed via a public IP, but password protected.
```

Comment:
Can we just use our existing prom/grafana setup?

Reply:
Yes. Local clusters used in CI/CD can deploy their own grafana/prom within the k8s cluster, and we can disable that for `staging` and update the existing/external prometheus to scrape from the opentel collector within the `staging` k8s cluster.